src/scala/ly/stealth/mesos/kafka/Rebalancer.scala

```diff
@@ -121,6 +144,36 @@ class Rebalancer {
     reassignment.toMap
   }
 
+  private def getRealignments(brokerIds: Seq[Int], topics: util.List[String], assignment: Map[TopicAndPartition, Seq[Int]], replicas: Int = -1, fixedStartIndex: Int = -1): Map[TopicAndPartition, Seq[Int]] = {
+    var realignment : Map[TopicAndPartition, Seq[Int]] = new mutable.HashMap[TopicAndPartition, List[Int]]()
+
+    val rand = new Random
+    var replacementReplicaIndex = if (fixedStartIndex != -1) fixedStartIndex else rand.nextInt(brokerIds.size)
+    assignment.foreach { entry =>
+      val topic = entry._1
+      val members = entry._2
+      val rf: Int = if (replicas != -1) replicas else members.size
+
+      if((rf != members.size) || ((brokerIds intersect members).size != members.size)) {
```

Comment:
Just realized this evening: this logic does handle shrinking replica counts where the broker ids are all permitted.  For example, if i allow 100-142, request replicas=3 and currently have members 100, 101, 102, 103, it will not properly shrink down to 100, 101, 102.  Will fix it in the AM.